File: v2/java/src/main/java/com/google/api/services/samples/adsense/cmdline/GetAllCustomChannels.java
/*
 * Copyright (c) 2021 Google LLC.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

package com.google.api.services.samples.adsense.cmdline;

import com.google.api.services.adsense.v2.Adsense;
import com.google.api.services.adsense.v2.model.CustomChannel;
import com.google.api.services.adsense.v2.model.ListCustomChannelsResponse;
import java.util.ArrayList;
import java.util.List;

/**
*
* This example gets all custom channels in an ad client.
*
* Tags: accounts.adclients.customchannels.list
*
*/
public class GetAllCustomChannels {

  /**
   * Runs this sample.
   *
   * @param adsense AdSense service object on which to run the requests.
   * @param adClientId the ID for the ad client to be used.
   * @param maxPageSize the maximum page size to retrieve.
   * @return the retrieved custom channels.
   * @throws Exception
   */
  public static List<CustomChannel> run(Adsense adsense, String adClientId,
      int maxPageSize) throws Exception {
    System.out.println("=================================================================");
    System.out.printf("Listing all custom channels for ad client %s\n", adClientId);
    System.out.println("=================================================================");

    // Retrieve custom channel list in pages and display the data as we receive it.
    String pageToken = null;
    List<CustomChannel> allCustomChannels = new ArrayList<CustomChannel>();
    do {
      ListCustomChannelsResponse response = adsense.accounts().adclients().customchannels()
          .list(adClientId)
          .setPageSize(maxPageSize)
          .setPageToken(pageToken)
          .execute();
      List<CustomChannel> customChannels = response.getCustomChannels();

      if (customChannels != null && !customChannels.isEmpty()) {
        allCustomChannels.addAll(customChannels);
        for (CustomChannel channel : customChannels) {
          System.out.printf("Custom channel with id \"%s\" and name \"%s\" was found.\n",
              channel.getName(), channel.getDisplayName());
        }
      } else {
        System.out.println("No custom channels found.");
      }

      pageToken = response.getNextPageToken();
    } while (pageToken != null);

    System.out.println();
    return allCustomChannels;
  }
}

File: v2/php/examples/GetAllCustomChannels.php
<?php
/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * This example gets all custom channels in an ad client.
 *
 * Tags: accounts.customchannels.list
 */
class GetAllCustomChannels {
  /**
   * Gets all custom channels in an ad client.
   *
   * @param $service Google_Service_AdSense AdSense service object on which to
   *     run the requests.
   * @param $adClientId string the ID for the ad client to be used.
   * @param $maxPageSize int the maximum page size to retrieve.
   * @return array the last page of retrieved custom channels.
   */
  public static function run($service, $adClientId, $maxPageSize) {
    $separator = str_repeat('=', 80) . "\n";
    print $separator;
    printf("Listing all custom channels for ad client %s\n", $adClientId);
    print $separator;

    $optParams['pageSize'] = $maxPageSize;

    $pageToken = null;
    $customChannels = null;
    do {
      $optParams['pageToken'] = $pageToken;
      $result = $service->accounts_adclients_customchannels
                        ->listAccountsAdclientsCustomchannels($adClientId,
                                                              $optParams);
      if (!empty($result['customChannels'])) {
        $customChannels = $result['customChannels'];
        foreach ($customChannels as $customChannel) {
          printf("Custom channel with ID \"%s\" and name \"%s\" was found.\n",
              $customChannel['name'], $customChannel['displayName']);
        }
        $pageToken = $result['nextPageToken'];
      } else {
        print "No custom channels found.\n";
      }
    } while ($pageToken);
    print "\n";

    return $customChannels;
  }
}
